SayTheirNames/Source/Dependency/Deep Linking/DeepLinkHandler.swift

```diff
@@ -36,25 +36,36 @@ final class DeepLinkHandler: Dependency {
     // MARK: - Public
     
     public func handle(urlContext: Set<UIOpenURLContext>) {
-        for context in urlContext {
-             if let deepLink = self.deepLink(matching: context) {
+        self.handle(urls: Set(urlContext.map { $0.url }))
+    }
+    
+    // We use this method as an entry point for unit tests.
+    // So we technically don't care about the resulting `DeepLink` array
+    // except for the unit tests.
+    @discardableResult
+    public func handle(urls: Set<URL>) -> [DeepLink] {
```

Comment:
This should never return more than one DeepLink.

Reply:
Agreed. @hybridcattt and I had that conversation earlier. Since the `scene(_:openURLContexts:)` provides a `Set` I wasn't sure if there would ever be a case where multiple `UIOpenURLContext` was relevant.

It's actually not a big deal for the unit tests because we have an assertion checking on the number of elements in the set. So I left the original code (iterating through the Set) as-is since I didn't want to change the initial intent of your code. But we can change that if you'd like.

Reply:
Note : if we decide to change this, let's do this in another PR so as to clearly point out the reason for the change (otherwise it will get drown in here).